app/controllers/home_controller.rb

```diff
@@ -1,6 +1,27 @@
 class HomeController < ApplicationController
-
+  before_action :card_sample, only: [:index, :success, :error]
+  
   def index
   end
 
+  def checking
+    @card = Card.find(params[:id])
+    if @card.check(params[:original_text]) == true
+      @card.update(params.permit(:original_text, :translated_text))
+      redirect_to '/success'
+    else
+      redirect_to '/error'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.